system-test/install.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as execa from 'execa';
+import * as mv from 'mv';
+import {ncp} from 'ncp';
+import * as tmp from 'tmp';
+import {promisify} from 'util';
+
+const keep = false;
+const mvp = promisify(mv) as {} as (...args: string[]) => Promise<void>;
+const ncpp = promisify(ncp);
+const stagingDir = tmp.dirSync({keep, unsafeCleanup: true});
+const stagingPath = stagingDir.name;
+const pkg = require('../../package.json');
+
+describe('📦 pack and install', () => {
+  /**
+   * Create a staging directory with temp fixtures used to test on a fresh
+   * application.
+   */
+  it('should be able to use the d.ts', async () => {
+    await execa('npm', ['pack', '--unsafe-perm']);
```

Comment:
I like this approach to integration testing the TypeScript module, I wonder if this could be be abstracted into a module.

Reply:
@JustinBeckwith I see that you chose not to use [post-install-check](https://npm.im/post-install-check). I am guessing that there are things in that module that you don't like. Can you work with @DominicKramer and provide feedback to make sure we are serving your needs and making that module better for everyone ❤️.

Reply:
Great call out.  Feedback left!
https://github.com/google/post-install-check/issues

Reply:
Oh, I wasn't trying to be a :trollface:, I didn't realize this was essentially what `post-install-check` does.

It would be nice to eventually consolidate on a module like `post-install-check` ... I honestly don't love the name `post-install-check` if what it does is check that TypeScript has the appropriate dependencies. I read `post-install-check` and assumed it was something more abstract (a generic way of checking your packages post-install scripts?). I'd advocate a name like `check-installed-types`?

Reply:
Yep, it is not the best name. We can rename it. How about something like `ts-pack-n-play`?

/cc @googleapis/node-team FYI

Reply:
FWIW - I love that name.  @ofrobots wanna open an issue over there to consider the rename?